### ISSUE TITLE
Any plans of converting this to ES6? :) :) :)

### ISSUE THREAD

**Issue by joshuairl** [bbc/tal#458]
Pweeeaasssseeeeeee..... xD

**Comment by subsidel** [comment]
Hi Joshua,

Thanks for your enthusiastic request!

Unfortunately, not all of the devices we support are capable of running es6/es2015 code.
This eliminates the easy option of just swapping out things for their es6/es2015 
equivalents.

We recently attempted an es6 => es5 transpilation variant of a subsection
of TAL, however, we did run into some issues with some of the older devices
we support.

In the investigation we found that the Babel ES2015 preset uses ES6 functions like
`Array.from` and would not transpile `Array.includes` to an ES5 implementation.
This is because it makes an assumption that you have included the ES2015 polyfill
and this polyfill did not work on a subset of the supported devices.

We also tried alternative polyfills such as [es6-shim](https://github.com/paulmillr/es6-shim), however since
these polyfills were >50KB, we had concerns about including such files by 
default, and how this affect application size and load times.

This shouldn't stop you/any other users of TAL using es6/es2015+ in your own code
if you only need to support certain newer devices. Then again YMMV. 

This does mean that for the time being we have no plans to convert the current implementation of TAL directly to es6/es2015.
